subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/annotations/NestedBeanAnnotationHandler.java

```diff
@@ -53,21 +53,24 @@ public boolean isPropertyRelevant() {
 
     @Override
     public boolean shouldVisit(PropertyVisitor visitor) {
-        return !visitor.visitOutputFilePropertiesOnly();
+        return true;
```

Comment:
I think we don't need `shouldVisit()` and `visitOutputFilePropertiesOnly()` at all now. It was never the "visiting" that was expensive, but the creation of the exceptions. And now we need to visit everything even if we are only looking for output properties only anyway.

Reply:
Good point.